internal/locate/region_request.go

```diff
@@ -113,7 +112,8 @@ type RegionRequestSender struct {
 	replicaSelector   ReplicaSelector
 	failStoreIDs      map[uint64]struct{}
 	failProxyStoreIDs map[uint64]struct{}
-	RegionRequestRuntimeStats
+	Stats             *RegionRequestRuntimeStats
```

Comment:
Maybe we could also collect them in memory and dump print logs timely in kv-client, so we could know how many number of region errors are returned from which specific tikv store.

Reply:
How about add a metric for it with labels  (region_error, store_id)?

Reply:
This is a straightforward way, I think we could add it first. 
The drawback of metric is that it does not contain the detailed informaiton as logs except the region errors.

Reply:
I think the metrics with labels (region_error, store_id) could be ok now.

 
Collect those info in memory and dump print logs timely in kv-client will Introduces additional complexity, and may also result in more logs being generated.

Reply:
Already add a metrics for region_err. And also add metric for rpc_error.

<img width="1439" alt="image" src="https://github.com/tikv/client-go/assets/26020263/0bdce115-0579-4d80-aa94-2a48492c638c">